Implement nickel_bluetooth action

onatbas opened the pull request:
Implementation of a new action "nickel_bluetooth" to turn Bluetooth on and off via NickelMenu.

Available commands are:
	enable
	disable
	toggle

I've been testing/using this modified version of NickelMenu on my Clara 2E and It should be compatible with all Bluetooth devices. 

This should address issue  #142 

Usage examples:
menu_item :main  :Bluetooth (enable)      :nickel_bluetooth     :enable
menu_item :main  :Bluetooth (disable)     :nickel_bluetooth     :disable
menu_item :main  :Bluetooth (toggle)      :nickel_bluetooth     :toggle

pgaskin commented on src/action_cc.cc:
```diff
@@ -896,3 +897,42 @@ NM_ACTION_(cmd_output) {
+    //_ZN16BluetoothManager14sharedInstanceEv
+    BluetoothManager *(*BluetoothManager_sharedInstance)();
+    NM_ACT_XSYM(BluetoothManager_sharedInstance, "_ZN16BluetoothManager14sharedInstanceEv", "could not dlsym BluetoothManager::sharedInstance");
+
+
```
Stray newline.

pgaskin commented on src/action_cc.cc:
```diff
@@ -896,3 +897,42 @@ NM_ACTION_(cmd_output) {
+        void (*BluetoothManager_on)(BluetoothManager *);
+        NM_ACT_XSYM(BluetoothManager_on, "_ZN16BluetoothManager2onEv", "could not dlsym BluetoothManager::on");
+        BluetoothManager_on(btm);
+    }else{
```
Formatting: spaces around else.

pgaskin commented on src/action_cc.cc:
```diff
@@ -896,3 +897,42 @@ NM_ACTION_(cmd_output) {
+    BluetoothManager *btm = BluetoothManager_sharedInstance();
+    NM_CHECK(nullptr, btm, "could not get shared bluetooth manager pointer");
+
+    if (toggle){
```
Formatting: add a space before the brace (same for the next if).

pgaskin reviewed (changes requested):
Look okay overall, not tested (I don't have a Kobo with bluetooth support).

A few minor formatting nitpicks (I should add clang-format at some point).

Also, add the `//libnickel` symbol checks with the version constraints for CI.

onatbas commented:
Thanks. I identified a few problems with the internal Bluetooth driver: in certain conditions, the connection doesn't get established if BT is turned on and off too quickly. I added a few actions and parameters to this fix this and currently testing the changes. 
In addition to addressing your comments, I shall submit my changes by the end of today.

onatbas commented:
I have completed my checks and happy with the current implementation of this feature. The //libnickel checks could probably go even earlier versions but, I haven’t had the chance to validate all versions. 
let me know if there are other changes, style adjustments I may have missed.

pgaskin reviewed (approved):
Looks perfect; I like how you organized the actions. I can't test it as I don't have one which supports it, but it's simple enough, I'll probably just merge it.

pgaskin commented on src/action_cc.cc:
```diff
@@ -896,3 +897,72 @@ NM_ACTION_(cmd_output) {
+    void (*BluetoothManager_scan)(BluetoothManager *);
+    NM_ACT_XSYM(BluetoothManager_scan, "_ZN16BluetoothManager4scanEv", "could not dlsym BluetoothManager::BluetoothManager::scanEv");
+
+    //libnickel 4.34.20097 *_ZN16BluetoothManager8stopScanEv
```
Oh, oops, didn't notice the missed space here.

onatbas replied:
My bad! Just sent the correction.

onatbas commented:
I'm glad you like it. It's an old pattern called "bitmasking" or "bit flags." The method was going toward multiple booleans, and this pattern is intended for that problem.

It is indeed very simple as you quite rightly put it. Here's a sample from my Clara 2E. (Bluetooth device blinks when it's  disconnected/scanning.  When the blinking stops, it means connection established.)  
https://github.com/pgaskin/NickelMenu/assets/714795/f6f404ee-7f92-45ad-90be-0bd51cd4ea35
